app/api/saml/acs/route.ts

```diff
@@ -0,0 +1,172 @@
+import { NextRequest, NextResponse } from 'next/server'
+import forge from 'node-forge'
+
+export async function POST(request: NextRequest) {
+  try {
+    // console.log('📨 SAML callback received')
+
+    const formData = await request.formData()
+    const samlResponse = formData.get('SAMLResponse') as string
+
+    if (!samlResponse) {
+      throw new Error('No SAML response received')
+    }
+
+    // console.log('🔍 Processing SAML response with decryption...')
+
+    // Decode the base64 SAML response
+    const decodedResponse = Buffer.from(samlResponse, 'base64').toString('utf-8')
```

Comment:
What wasn't working with the samilfy lib? You're doing a lot of manual parsing that should be handled by the library.